test/OpenFeature.IntegrationTests/Steps/EvaluationStepDefinitions.cs

```diff
@@ -0,0 +1,275 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.DataAnnotations;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using OpenFeature.Constant;
+using OpenFeature.Contrib.Providers.Flagd;
+using OpenFeature.Model;
+using TechTalk.SpecFlow;
+using Xunit;
+
+namespace OpenFeature.IntegrationTests
+{
+    [Binding]
+    public class EvaluationStepDefinitions
+    {
+        private readonly ScenarioContext _scenarioContext;
+        private static FeatureClient client;
+        private Task<bool> booleanFlagValue;
+        private Task<string> stringFlagValue;
+        private Task<int> intFlagValue;
+        private Task<double> doubleFlagValue;
+        private Task<Value> objectFlagValue;
+        private Task<FlagEvaluationDetails<bool>> booleanFlagDetails;
+        private Task<FlagEvaluationDetails<string>> stringFlagDetails;
+        private Task<FlagEvaluationDetails<int>> intFlagDetails;
+        private Task<FlagEvaluationDetails<double>> doubleFlagDetails;
+        private Task<FlagEvaluationDetails<Value>> objectFlagDetails;
+        private string contextAwareFlagKey;
+        private string contextAwareDefaultValue;
+        private string contextAwareValue;
+        private EvaluationContext context;
+        private string notFoundFlagKey;
+        private string notFoundDefaultValue;
+        private FlagEvaluationDetails<string> notFoundDetails;
+        private string typeErrorFlagKey;
+        private int typeErrorDefaultValue;
+        private FlagEvaluationDetails<int> typeErrorDetails;
+
+        public EvaluationStepDefinitions(ScenarioContext scenarioContext)
+        {
+            _scenarioContext = scenarioContext;
+            var flagdProvider = new FlagdProvider();
+            Api.Instance.SetProvider(flagdProvider);
+            client = Api.Instance.GetClient();
+        }
+
+        [Given(@"a provider is registered with cache disabled")]
+        public void Givenaproviderisregisteredwithcachedisabled()
+        {
+            //_scenarioContext.Pending();
```

Comment:
is this needed?

Reply:
the method yes, the commented code no, will remove :)